shoop/admin/modules/contacts/views/detail.py

```diff
@@ -8,19 +8,23 @@
 from __future__ import unicode_literals
 
 from django.conf import settings
+from django.contrib import messages
 from django.core.urlresolvers import reverse
 from django.db.models import Q
+from django.http.response import HttpResponseRedirect
 from django.utils.encoding import force_text
 from django.utils.translation import ugettext_lazy as _
 from django.views.generic import DetailView
 
 from shoop.admin.toolbar import PostActionButton, Toolbar, URLActionButton
 from shoop.core.models import CompanyContact, Contact, Order, PersonContact
+from shoop.utils.excs import Problem
 
 
 class ContactDetailToolbar(Toolbar):
-    def __init__(self, contact):
+    def __init__(self, contact, path):
```

Comment:
Instead of passing the path. Pass the request. The path is too specifig and the request might come handy in some point.